framework/src/modules/chain/modules/blocks/verify.js

```diff
@@ -835,9 +835,10 @@ Verify.prototype.processBlock = function(block, broadcast, saveBlock, cb) {
 			// 'false' if block comes from chain synchronisation process
 			updateSystemHeaders(seriesCb) {
 				// Update our own headers: broadhash and height
-				!library.config.loading.snapshotRound
-					? components.system.update(seriesCb)
-					: seriesCb();
+				if (!library.config.loading.snapshotRound) {
+					return components.system.update().then(() => setImmediate(seriesCb));
+				}
+				return setImmediate(seriesCb);
```

Comment:
This could be simplified this way:
```
if (!library.config.loading.snapshotRound) {
  await components.system.update();
}
return setImmediate(seriesCb);
```

Reply:
Same as above.